pynxtools/dataconverter/writer.py

```diff
@@ -306,4 +316,28 @@ def add_units_key(dataset, path):
                 raise IOError(f"Unknown error occured writing the path: {path} "
                               f"with the following message: {str(exc)}") from exc
 
-        self.output_nexus.close()
+    def write(self):
+        """Writes the NeXus file with previously validated data from the reader with NXDL attrs."""
+        try:
+            if self.write_in_memory:
+                raise ValueError("To write in memory and get the file obhect please use  "
+                                 "the method get_in_memory_obj()")
+            self._put_data_into_hdf5()
+        finally:
+            self.output_nexus.close()
+
+    def get_in_memory_obj(self):
+        """Write the nexus file as in-memory obj.
+
+        Write in nexus in-memory file obj and return that file object.
+        """
+        try:
+            if self.write_in_memory:
+                self._put_data_into_hdf5()
+            else:
+                raise ValueError("The write_in_memory variable is False Writer"
+                                 "class initialization.")
+        except Exception as excp:
+            self.output_nexus.close()
+            raise excp
+        return self.output_nexus
```

Comment:
I don't understand why do we need these two separate functions. Aren't we using the h5py backend to be in memory so the interface being written to in this file shouldn't change?

Reply:
I am not sure I understood about `using h5py in backend`. The function `get_in_memory_obj` is not used in pynxtools. get_in_memory_obj used to populate nexus section in nomad without writing the data in nexus file.

Reply:
If you change the way you open the file using h5py.File, then you don't need to have separate functions for this. That open statement should decide whether it's writing in memory or disk.  Having these two routes just seems unnecessary.